### PR TITLE
added new form type

### PR DESCRIPTION
**Pull request by Deepak-Bagrawat**
https://support.cop.homeoffice.gov.uk/browse/EAB-629

### CODE REVIEW

**Review by gardnersam — COMMENTED** [review]
It seems like FORMWITHTASK is the same as TASK, all the points it's been added in they do the same thing. Are there more PRs to come? I'm not sure how this alone would let you have pages before a tasklist?

File: src/models/FormTypes.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const TYPE_HUB_AND_SPOKE = 'hub-and-spoke';
 const TYPE_TASK_LIST = 'task-list';
 const TYPE_WIZARD = 'wizard';
 const TYPE_TASK_LIST_CYA = 'task-list-cya';
+const Type_FORM_WITH_TASK_LIST = 'form-with-task-list';
```
Comment (paulfgraham) [inline src/models/FormTypes.js:7]: Inconsistent case with the others, and also not alphabetical.

File: src/models/FormTypes.js [hunk 2]
```diff
@@ -12,6 +13,7 @@ const FormTypes = {
   TASK: TYPE_TASK_LIST,
   WIZARD: TYPE_WIZARD,
   TASKCYA: TYPE_TASK_LIST_CYA,
+  FORMWITHTASK: Type_FORM_WITH_TASK_LIST,
```
Comment (paulfgraham) [inline src/models/FormTypes.js:16]: Not alphabetical. Also, shouldn't this be SNAKE_UPPER_CASE? The same as the `TASK_CYA` one really... but that horse appears to have bolted.

**Review by paulfgraham — CHANGES_REQUESTED** [review]

File: src/components/FormRenderer/helpers/getSubmissionStatus.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { FormPages, FormTypes, PageAction } from '../../../models';
 import getNextPageId from './getNextPageId';
 
 const getSubmissionStatus = (formType, pages, currentPageId, action, formData, currentTask, isCompleted) => {
-  if (formType === FormTypes.TASK) {
+  if (formType === FormTypes.TASK || formType === FormTypes.FORMWITHTASK) {
```
Comment (paulfgraham) [inline src/components/FormRenderer/helpers/getSubmissionStatus.js:5]: This should surely be `FormTypes.FORM_WITH_TASK` now, shouldn't it?

File: src/components/FormRenderer/helpers/getSubmissionStatus.test.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ describe('components', () => {
         describe(`when the action type is '${PageAction.TYPES.SAVE_AND_RETURN}'`, () => {
 
           Object.values(FormTypes).forEach(formType => {
-            if(formType !== FormTypes.TASK){
+            if(formType !== FormTypes.TASK && formType !== FormTypes.FORMWITHTASK ){
```
Comment (paulfgraham) [inline src/components/FormRenderer/helpers/getSubmissionStatus.test.js:59]: Same here... this should be `FormTypes.FORM_WITH_TASK` now, shouldn't it?

File: src/models/FormTypes.js [hunk 5]
```diff
@@ -4,14 +4,16 @@ const TYPE_HUB_AND_SPOKE = 'hub-and-spoke';
 const TYPE_TASK_LIST = 'task-list';
 const TYPE_WIZARD = 'wizard';
 const TYPE_TASK_LIST_CYA = 'task-list-cya';
+const TYPE_FORM_WITH_TASK_LIST = 'form-with-task-list';
```
Comment (paulfgraham) [inline src/models/FormTypes.js:7]: This is still not alphabetical.

File: src/utils/Hub/getFormHub.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ const getFormHub = (type, hub, components, formData) => {
       return FormPage.get(hub, components, formData);
     }
   }
-  if (type === FormTypes.TASK) {
+  if (type === FormTypes.TASK || type === FormTypes.FORMWITHTASK) {
```
Comment (paulfgraham) [inline src/utils/Hub/getFormHub.js:27]: Again... shouldn't this be `FormTypes.FORM_WITH_TASK` now?

**Review by paulfgraham — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@ukhomeoffice/cop-react-form-renderer",
-  "version": "3.15.0",
+  "version": "3.16.0",
```
Comment (paulfgraham) [inline package.json:3]: Oh, there's already a `3.16.0`... so this is out of date with the `main` branch.

**Review by paulfgraham — APPROVED** [review]

**Pull request merged**